### PR TITLE
Manage users

### PR DESCRIPTION
**Pull request by yimialmonte**
Create an endpoint for register users.

URI: http://localhost:3000/v1/users/register
Body: 
{
    "name": "juan",
    "email": "juan@gma.com",
    "password": "jufhfhfhfgh"
}

### CODE REVIEW

File: components/users/userController.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import User from './user'
+
+const register = async(req, res) => {
+  try {
+    console.log(req.body)
```
Comment (cesar18pena) [inline components/users/userController.js:5]: Remove this `console.log` from the code

File: config/database.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import mongoose from 'mongoose'
+
+export default function db() {
+  mongoose.connect('mongodb://127.0.0.1:27017/smart-offer', {
```
Comment (cesar18pena) [inline config/database.js:4]: It is ok for now to have this hardcode string, but remember we need to add support and use `environment variables`

File: components/users/userController.js (hunk 1 above)
Comment (yimialmonte) [reply components/users/userController.js]: Removed

File: config/database.js (hunk 2 above)
Comment (yimialmonte) [reply config/database.js]: Fixed, now I am using environment variable

File: .env [hunk 3]
```diff
@@ -0,0 +1 @@
+DATABASE_URL = 'mongodb://127.0.0.1:27017/smart-offer'
```
Comment (cesar18pena) [inline .env:1]: You should never upload the `env` file in your Github project, add this your `.gitignore`

Comment (yimialmonte) [reply .env]: Removed from the Repo, thanks

File: components/users/user.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import mongoose from 'mongoose'
+import bcryptjs from 'bcryptjs'
```
Comment (cesar18pena) [inline components/users/user.js:2]: Considering this application would be only in NodeJS, we can use `bcrypt` instead of `bcryptjs`. You can read more about it: https://stackoverflow.com/questions/54674387/how-do-nodes-bcrypt-and-bcryptjs-libraries-differ.

We can do it in another PR, but for you to know

File: components/users/userValidations.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import * as Yup from 'yup'
+import User from './user'
+
+const RegisterSchema = Yup.object().shape({
+  name: Yup.string().required(),
+  email: Yup.string().email().required(),
+  password: Yup.string().min(6).required()
```
Comment (cesar18pena) [inline components/users/userValidations.js:7]: Probably we want to make a password an `8` digits required same as 90% of pages

File: env/index.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import dotenv from 'dotenv'
+
+dotenv.config()
+
+export default {
+  databaseurl: process.env.DATABASE_URL || 'mongodb://127.0.0.1:27017/smart-offer'
```
Comment (cesar18pena) [inline env/index.js:6]: Change this to `databaseUrl`; remember to follow up `camelCase`

Comment (yimialmonte) [reply env/index.js]: Fixed
